### PR TITLE
Add login to registry before pull image.

### PR DESCRIPTION
**Pull request by huy-axie**

### CODE REVIEW

File: libraries/docker/steps/build.groovy [hunk 1]
```diff
@@ -8,18 +8,42 @@ import com.cloudbees.plugins.credentials.Credentials
...
-      }
+        if (config.path_dockerfile != "") {
+          // check if base image is private
+          base = sh (
+            script: """cat $config.path_dockerfile | grep FROM | grep gcr""",
+            returnStdout: true
+          ).trim()
+
+          // check if base not null -> this is a private image, do login to registry
+          if (base?.trim()){
```
Comment (huyngopt1994) [inline libraries/docker/steps/build.groovy:19]: do we need trim again ?

File: libraries/docker/steps/build.groovy [hunk 2]
```diff
@@ -8,18 +8,42 @@ import com.cloudbees.plugins.credentials.Credentials
...
+              """
+            }
+          // if base is null -> this is not a private image
+          }else {
+            sh """
+              base_images=`cat $config.path_dockerfile | grep FROM | awk '{print \$2}'`
+              for i in \$base_images
+              do
+                img_id=`docker images -q \$i`
+                if [[ -z \$img_id ]]; then
```
Comment (huyngopt1994) [inline libraries/docker/steps/build.groovy:39]: could we avoid duplicating this logic like put it into a function ?

**Review by huyngopt1994 — APPROVED** [review]

**Comment by huyngopt1994** [comment]
LGTM

File: libraries/docker/steps/build.groovy (hunk 1 above)
Comment (huy-axie) [reply libraries/docker/steps/build.groovy:19]: yes, failed without `trim()`.

**Pull request merged**